packages/number-input/src/Input/Input.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { ChangeEvent, KeyboardEvent, useEffect, useState } from 'react';
+
+import { cx } from '@leafygreen-ui/emotion';
+import { useControlledValue, useForwardedRef } from '@leafygreen-ui/hooks';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import { createSyntheticEvent } from '@leafygreen-ui/lib';
+import { transitionDuration } from '@leafygreen-ui/tokens';
+
+import { Arrows } from '../Arrows';
+import { ErrorIcon } from '../ErrorIcon';
+import { Direction, Size, State } from '../NumberInput/NumberInput.types';
+
+import {
+  inputAnimateStyles,
+  inputBaseStyles,
+  inputErrorAnimateStyles,
+  inputErrorPaddingTransitionStyles,
+  inputThemeStyles,
+  selectBaseStyles,
+  sizeInputStyles,
+  wrapperBaseStyles,
+  wrapperClassName,
+  wrapperDisabledStyles,
+  wrapperHoverStyles,
+  wrapperStateStyles,
+  wrapperThemeStyles,
+} from './Input.styles';
+import { InputProps } from './Input.types';
+
+/**
+ * @internal
+ */
+
+export const Input = React.forwardRef<HTMLInputElement, InputProps>(
+  (
+    {
+      value: valueProp,
+      onChange: onChangeProp,
+      disabled = false,
+      size = Size.Default,
+      state = State.None,
+      hasSelectOptions,
+      errorMessage,
+      className,
+      ...rest
+    }: InputProps,
+    forwardRef,
+  ) => {
+    /**
+     * Timeout for error icon transitions
+     */
+    let translateTimeout: NodeJS.Timeout;
+    const [shouldErrorTransition, setShouldErrorTransition] =
+      useState<boolean>(false);
+    const inputRef = useForwardedRef<HTMLInputElement | null>(forwardRef, null);
+    const { theme } = useDarkMode();
+
+    useEffect(() => {
+      return () => {
+        clearTimeout(translateTimeout);
+      };
+    });
```

Comment:
does this need deps?

Reply:
i'm not getting any errors

Reply:
no errors because this means it'll just run once but I don't think that's the behavior you want, right?

Reply:
I think that the cleanup is performed when the component unmounts

Reply:
https://legacy.reactjs.org/docs/hooks-effect.html#example-using-hooks-1

> When exactly does React clean up an effect? React performs the cleanup when the component unmounts.

Reply:
But do you want this only performed when the component unmounts? that's what I don't understand

Reply:
Do you see something that I might be over looking?

Reply:
ah ok sorry you're right